src/js/utils/defaults.ts

```diff
@@ -1,8 +1,9 @@
+import { GenericObject } from '../types/generic.type';
+
 // Adds properties to the first object from the rest
 // Does not add properties which exist anywhere in the object or it's prototype chain (no shadowing, no overriding)
-
-export default function defaults(obj) {
-    Array.prototype.slice.call(arguments, 1).forEach(function(source) {
+export default function defaults(obj: GenericObject, ...rest: any[]): GenericObject {
```

Comment:
are we polyfilling rest spread?

Reply:
Babel handles it. Recent eslint rule changes don't give me a choice:
https://github.com/jwplayer/jwplayer/commit/befe9f10da2f171b3d51473887633968eb8ff65f
https://github.com/jwplayer/jwplayer/commit/7404a3e284b094e8a70b870826cf46ed48abf980
```
"prefer-rest-params": "error",
"prefer-spread": "error",
```

Reply:
Can remove those if wanted. I took those rules from the recommended typescript-eslint rules list.

Reply:
Rest spreads are great! Doesn't this looklbetter?

Reply:
It is beautiful